### PR TITLE
use lock objects on the threads run methods

### PR DESCRIPTION
**Pull request by ralphiee22**
### Summary

It seems that the vacuum and ping command are trying to access the database at the same time during startup. This causes some important models to not be generated by the ping command, as well as database is locked errors. Although, kolibri starts per usual even given this error. This PR adds a blocking lock to the threads so one process can not execute until the other has finished.

### Reviewer guidance

`make assets`
`kolibri start`
is there any database is locked errors?

### References

https://github.com/learningequality/kolibri/issues/4334

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/utils/server.py [hunk 1]
```diff
@@ -46,6 +46,9 @@
 # Currently non-configurable until we know how to properly handle this
 LISTEN_ADDRESS = "0.0.0.0"
 
+# use locks so vacuum doesn't conflict with ping
+lock = threading.Lock()
```
Comment (benjaoming) [inline kolibri/utils/server.py]: Maybe rename this lock to reflect its intention. What is it locking?

File: kolibri/utils/server.py [hunk 2]
```diff
@@ -115,8 +119,9 @@ def start_command(cls):
         thread.start()
 
     def run(self):
-        # Try to do the vacuum every 3 hours
-        call_command("vacuumsqlite", interval=10)
+        with lock:
+            # Try to do the vacuum every 3 hours
```
Comment (benjaoming) [inline kolibri/utils/server.py]: Wondering if this comment makes sense when it says `interval=10` ?

File: kolibri/utils/server.py [hunk 3]
```diff
@@ -115,8 +119,9 @@ def start_command(cls):
...
+            call_command("vacuumsqlite", interval=10)
```
Comment (benjaoming) [inline kolibri/utils/server.py]: Isn't it the `vacuumsqlite` command that does the blocking wait? In that case, it should have the lock as an argument in the call or import it from `kolibri.utils.server`?

Comment (benjaoming) [reply kolibri/utils/server.py]: ...otherwise we risk a deadlock because it never yields the lock for the other waiting thread?

**Review by benjaoming — CHANGES_REQUESTED** [review]
I like the solution with locks, those are straight-forward.

Regarding testing: Just like the error showed up, perhaps merging this as is would reveal if it was the sole cause.

File: kolibri/utils/server.py (hunk 2 above)
Comment (ralphiee22) [reply kolibri/utils/server.py]: planning to change that according to the tagged issue

**Review by benjaoming — APPROVED** [review]
Regarding testing: Just like the error showed up, perhaps merging this as is would reveal if it was the sole cause.

File: kolibri/utils/server.py [hunk 4]
```diff
@@ -116,7 +119,7 @@ def start_command(cls):
 
     def run(self):
         # Try to do the vacuum every 3 hours
-        call_command("vacuumsqlite", interval=10)
+        call_command("vacuumsqlite", interval=True)
```
Comment (indirectlylit) [inline kolibri/utils/server.py]: can we rename this from `interval` to `scheduled`, `repeat`, or something similar?

**Review by indirectlylit — APPROVED** [review]
thank you!

**Pull request merged**